evalml/automl/engine/engine_base.py

```diff
@@ -91,21 +91,27 @@ def submit_scoring_job(self, automl_config, pipeline, X, y, objectives):
         """Submit job for pipeline scoring."""
 
 
-def train_pipeline(pipeline, X, y, optimize_thresholds, objective):
+def train_pipeline(pipeline, X, y, optimize_thresholds, objective, X_schema=None, y_schema=None):
     """Train a pipeline and tune the threshold if necessary.
 
     Arguments:
         pipeline (PipelineBase): Pipeline to train.
-        X (ww.DataTable, pd.DataFrame): Features to train on.
-        y (ww.DataColumn, pd.Series): Target to train on.
+        X (pd.DataFrame): Features to train on.
+        y (pd.Series): Target to train on.
         optimize_thresholds (bool): Whether to tune the threshold (if pipeline supports it).
         objective (ObjectiveBase): Objective used in threshold tuning.
+        X_schema (ww.TableSchema): Woodwork schema.
+        y_schema (ww.ColumnSchema): Woodwork schema.
 
     Returns:
         pipeline (PipelineBase): trained pipeline.
     """
     X_threshold_tuning = None
     y_threshold_tuning = None
+    if X_schema:
+        X.ww.init(schema=X_schema)
+    if y_schema:
+        y.ww.init(schema=y_schema)
```

Comment:
Nice, so this sets the schema and avoids us having to infer feature types from scratch